snapcraft/tests/test_commands_build.py

```diff
@@ -50,14 +50,25 @@ def make_snapcraft_yaml(self, n=1):
         parts = []
         for i in range(n):
             part_dir = os.path.join(common.get_partsdir(), 'build{}'.format(i))
-            state_file = os.path.join(part_dir, 'state')
+            state_dir = os.path.join(part_dir, 'state')
             parts.append({
                 'part_dir': part_dir,
-                'state_file': state_file,
+                'state_dir': state_dir,
             })
 
         return parts
 
+    def verify_state(self, part_name, state_dir, expected_step):
+        self.assertTrue(os.path.isdir(state_dir),
+                        'Expected state directory for {}'.format(part_name))
+
+        # Expect every step up to and including the specified one to be run
+        index = common.COMMAND_ORDER.index(expected_step)
+        for step in common.COMMAND_ORDER[:index+1]:
+            self.assertTrue(os.path.exists(os.path.join(state_dir, step)),
+                            'Expected {!r} to be run for {}'.format(step,
```

Comment:
Ah, I'm sorry @sergiusens you keep asking me to do that and I keep forgetting :frowning: . Fixed!